grails-app/services/com/unifina/service/StreamService.groovy

```diff
@@ -38,27 +49,42 @@ class StreamService {
 		return Stream.findByUiChannelPath(uiChannelPath)
 	}
 
-	Stream createStream(Map params, User user, String id = IdGenerator.getShort()) {
-		Stream stream = new Stream(params)
-		stream.id = id
-		stream.config = params.config
-		if (stream.name == null || stream.name.trim() == "") {
-			stream.name = Stream.DEFAULT_NAME
-		}
+	Stream createStream(CreateStreamCommand cmd, User user) {
+		return createStream(cmd, user, null, null, true)
+	}
 
-		Map config = stream.getStreamConfigAsMap()
-		if (!config.fields) {
-			config.fields = []
+	Stream createStream(CreateStreamCommand cmd, User user, String uiChannelPath, Canvas uiChannelCanvas, boolean validateIDField) {
+		Stream stream = new Stream(
+			name: ((cmd.name == null || cmd.name.trim() == "")) ? Stream.DEFAULT_NAME : cmd.name,
+			description: cmd.description,
+			config: JSONUtil.createGsonBuilder().toJson(Stream.normalizeConfig(cmd.config)),
+			partitions: cmd.partitions,
+			uiChannel: cmd.uiChannel,
+			requireSignedData: cmd.requireSignedData,
+			requireEncryptedData: cmd.requireEncryptedData,
+			uiChannelPath: uiChannelPath,
+			uiChannelCanvas: uiChannelCanvas
+		)
+		if (cmd.id != null) {
+			if (validateIDField && !CustomStreamIDValidator.validate(cmd.id)) {
+				throw new ValidationException(new FieldError("stream", "id", null))
```

Comment:
Is it possible to attach an human-readable explanation / message to the `ValidationException`  explaining what went wrong? When the Exception gets eventually rendered as a JSON message, it could be useful for the end-user to know what went wrong.

Reply:
The error messages in the JSON:
`{
  "code": "VALIDATION_ERROR",
  "message": "Validation failed for fields:\nid (null)\n"
}`

So it says that the problem is in ID field, and something in the format of that field. We validate the ID with a regex, and therefore we don't know how the regex doesn't match. @harbu Would you like to get very detailed info like "Stream id contains an invalid character ~"?

Reply:
Created a separate ticket for this: BACK-78